Sources/AST/ASTVisitor/ASTVisitor.swift

```diff
@@ -805,6 +811,29 @@ public struct ASTVisitor {
                          passContext: postProcessResult.passContext)
   }
 
+  func visit(_ externalCall: ExternalCall, passContext: ASTPassContext) -> ASTPassResult<ExternalCall> {
+    var processResult = pass.process(externalCall: externalCall, passContext: passContext)
+
+    processResult.passContext.isExternalConfigurationParam = true
+    processResult.element.configurationParameters = processResult.element.configurationParameters.map { param in
+      let x = visit(param, passContext: processResult.passContext)
+      return processResult.combining(x)
+    }
+    processResult.passContext.isExternalConfigurationParam = false
+
+    processResult.passContext.isExternalCall = true
```

Comment:
how does this handle nested external calls? Since there is no stack to speak of (unless the contexts basically act as the stack) won't we prematurely unset this boolean at the end of the visit if we are visiting external calls that are nested?

Reply:
By nested external calls, do you mean something like `call! alpha.func(param: call! beta.func())` ? If that's the case, this should not work at all, since you should only be able to return a value from external calls using the `if let ... = call? ...` syntax.

Reply:
We now support parsing nested calls